validator/project_validator.go

```diff
@@ -668,7 +707,7 @@ func ensureReferentialIntegrity(project *model.Project, distroIDs []string, dist
 			}
 
 			for _, distro := range task.RunOn {
```

Comment:
I would rename `distro` to something more generic like `name` since distros and containers are mutually exclusive concepts.